src/components/importwms/ImportWmsDirective.js

```diff
@@ -291,30 +314,69 @@ goog.require('ga_urlutils_service');
       var view = map.getView();
       var mapSize = map.getSize();
 
-      // If a minScale is defined
-      if (layer.MaxScaleDenominator && extent) {
-
-        // We test if the layer extent specified in the
-        // getCapabilities fit the minScale value.
-        var layerExtentScale = getScaleFromExtent(view, extent, mapSize);
-
-        if (layerExtentScale > layer.MaxScaleDenominator) {
-          var layerExtentCenter = ol.extent.getCenter(extent);
-          var factor = layerExtentScale / layer.MaxScaleDenominator;
-          var width = ol.extent.getWidth(extent) / factor;
-          var height = ol.extent.getHeight(extent) / factor;
-          extent = [
-            layerExtentCenter[0] - width / 2,
-            layerExtentCenter[1] - height / 2,
-            layerExtentCenter[0] + width / 2,
-            layerExtentCenter[1] + height / 2
-          ];
-
-          var res = view.constrainResolution(
-              view.getResolutionForExtent(extent, mapSize), 0, -1);
-          view.setCenter(layerExtentCenter);
-          view.setResolution(res);
-          return;
+      var mapExtent = map.previousExtent_;
+      var x1 = mapExtent[0];
+      var y1 = mapExtent[1];
+      var x2 = mapExtent[2];
+      var y2 = mapExtent[3];
+
+      var layerExtentCenter = ol.extent.getCenter(extent);
+      var LCx = layerExtentCenter[0];
+      var LCy = layerExtentCenter[1];
+
+      // If the center of the layer extent is out of the map-extent-dx or
+      // map-extent-dy, the map extent is set as extent for the layer zoom in
+      if ((LCx < x1) || (LCx > x2) || (LCy < y1) || (LCy > y2)) {
```

Comment:
You can use `ol.extent.containsXY` , see the [api](http://openlayers.org/en/master/apidoc/ol.extent.html)